src/main/java/com/xpn/xwiki/plugin/lucene/LucenePlugin.java

```diff
@@ -115,13 +110,13 @@
    * The Lucene text analyzer, can be configured in <tt>xwiki.cfg</tt> using the key
    * {@link #PROP_ANALYZER} ( <tt>xwiki.plugins.lucene.analyzer</tt>).
    */
-  private Analyzer analyzer;
+  private volatile Analyzer analyzer;
 
   /**
    * Lucene index updater. Listens for changes and indexes wiki documents in a separate
    * thread.
    */
-  private IndexUpdater indexUpdater;
+  IndexUpdater indexUpdater;
```

Comment:
indexUpdater must be volatile or all accesses (flushCache and getters) must be synchronized.